### PR TITLE
GEODE-8043 - Created JBossModuleService and implemented loadModule.

### PR DESCRIPTION
**Pull request by yozaner1324**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-assembly/src/integrationTest/resources/expected_jars.txt [hunk 1]
```diff
@@ -75,6 +76,8 @@ lucene-queries
 lucene-queryparser
 mapstruct
 micrometer-core
+module1.jar
+module2.jar
```
Comment (kohlmu-pivotal) [inline geode-assembly/src/integrationTest/resources/expected_jars.txt:80]: why are these showing up here?

File: geode-assembly/build.gradle [hunk 2]
```diff
@@ -577,6 +579,9 @@ distributions {
         from {project(':geode-core').tasks.named('raJar')}
         from {project(':geode-core').tasks.named('jcaJar')}
 
+        from project(':geode-modules').tasks.named('module1Jar')
+        from project(':geode-modules').tasks.named('module2Jar')
```
Comment (kohlmu-pivotal) [inline geode-assembly/build.gradle:583]: I don't think it should exposed outside of the original module

File: geode-modules/src/main/java/org/apache/geode/services/module/impl/JBossModuleService.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+  @Override
+  public <T> List<T> loadService(String serviceName) {
+    return null;
+  }
+
+  @Override
+  public boolean unloadService(String serviceName) {
+    return false;
+  }
```
Comment (kohlmu-pivotal) [inline geode-modules/src/main/java/org/apache/geode/services/module/impl/JBossModuleService.java:135]: possibly we need to remove these add them into their respective JIRA

File: geode-assembly/src/integrationTest/resources/expected_jars.txt (hunk 1 above)
Comment (yozaner1324) [reply geode-assembly/src/integrationTest/resources/expected_jars.txt]: Removed.

File: geode-assembly/build.gradle (hunk 2 above)
Comment (yozaner1324) [reply geode-assembly/build.gradle]: Removed.

File: geode-modules/src/main/java/org/apache/geode/services/module/impl/JBossModuleService.java (hunk 3 above)
Comment (yozaner1324) [reply geode-modules/src/main/java/org/apache/geode/services/module/impl/JBossModuleService.java]: Removed.

**Pull request merged**